### PR TITLE
Update basic diagnostic capitalization/punctuation

### PR DESCRIPTION
**Pull request by jonmeow**
This is a primarily automated change:

- Search & replace for capitalization
    - `(CARBON_DIAGNOSTIC\((?:\n\s+)?\w+,(?:\n\s+)?\s\w+,(?:\n\s+)?\s")([A-Z])`
    - `$1\L$2`
- Search & replace for period
    - `(CARBON_DIAGNOSTIC\((?:\n\s+)?\w+,(?:\n\s+)?\s\w+,(?:\n\s+)?\s"(?:[^)]|\n)+)\.("[,)])`
    - `$1$2`
- Limited search & replace for `ERROR: ` -> `error: ` in streamed things
    - Leaving a TODO for command_line because there's more cleanup that can be done there
- Modify diagnostic_consumer.cpp
    - ERROR -> error
    - WARNING -> warning

### CODE REVIEW

File: toolchain/check/check.cpp [hunk 1]
```diff
@@ -1140,13 +1140,13 @@ static auto BuildApiMapAndDiagnosePackaging(
...
           CARBON_DIAGNOSTIC(DuplicateLibraryApi, Error,
-                            "Library's API previously provided by `{0}`.",
+                            "library's API previously provided by `{0}`",
                             std::string);
           unit_info.emitter.Emit(packaging->names.node_id, DuplicateLibraryApi,
                                  prev_filename.str());
         } else {
           CARBON_DIAGNOSTIC(DuplicateMainApi, Error,
-                            "Main//default previously provided by `{0}`.",
+                            "main//default previously provided by `{0}`",
```
Comment (zygoloid) [inline toolchain/check/check.cpp]:
```suggestion
                            "`Main//default` previously provided by `{0}`",
```

File: toolchain/check/check.cpp [hunk 2]
```diff
@@ -1270,8 +1270,8 @@ auto CheckParseTrees(
             } else {
               // The import hasn't been checked, indicating a cycle.
               CARBON_DIAGNOSTIC(ImportCycleDetected, Error,
-                                "Import cannot be used due to a cycle. Cycle "
-                                "must be fixed to import.");
+                                "import cannot be used due to a cycle. Cycle "
```
Comment (zygoloid) [inline toolchain/check/check.cpp]:
```suggestion
                                "import cannot be used due to a cycle; cycle "
```

File: toolchain/check/convert.cpp [hunk 3]
```diff
@@ -535,8 +535,8 @@ static auto ConvertStructToClass(Context& context, SemIR::StructType src_type,
   auto& dest_class_info = context.classes().Get(dest_type.class_id);
   if (dest_class_info.inheritance_kind == SemIR::Class::Abstract) {
     CARBON_DIAGNOSTIC(ConstructionOfAbstractClass, Error,
-                      "Cannot construct instance of abstract class. "
-                      "Consider using `partial {0}` instead.",
+                      "cannot construct instance of abstract class. "
+                      "Consider using `partial {0}` instead",
```
Comment (zygoloid) [inline toolchain/check/convert.cpp]:
```suggestion
                      "cannot construct instance of abstract class; "
                      "consider using `partial {0}` instead",
```

File: toolchain/check/handle_class.cpp [hunk 4]
```diff
@@ -346,13 +346,13 @@ static auto DiagnoseClassSpecificDeclRepeated(Context& context,
                                               SemIRLoc prev_loc,
                                               Lex::TokenKind tok) -> void {
   CARBON_DIAGNOSTIC(ClassSpecificDeclRepeated, Error,
-                    "Multiple `{0}` declarations in class.{1}", Lex::TokenKind,
+                    "multiple `{0}` declarations in class.{1}", Lex::TokenKind,
                     std::string);
   const llvm::StringRef extra = tok == Lex::TokenKind::Base
                                     ? " Multiple inheritance is not permitted."
```
Comment (zygoloid) [inline toolchain/check/handle_class.cpp]:
```suggestion
                    "multiple `{0}` declarations in class{1}", Lex::TokenKind,
                    std::string);
  const llvm::StringRef extra = tok == Lex::TokenKind::Base
                                    ? "; multiple inheritance is not permitted"
```
(or revert the changes to this file and merge #4327).

File: toolchain/check/handle_class.cpp [hunk 5]
```diff
@@ -454,8 +454,8 @@ constexpr BaseInfo BaseInfo::Error = {.type_id = SemIR::TypeId::Error,
 static auto DiagnoseBaseIsFinal(Context& context, Parse::NodeId node_id,
                                 SemIR::TypeId base_type_id) -> void {
   CARBON_DIAGNOSTIC(BaseIsFinal, Error,
-                    "Deriving from final type `{0}`. Base type must be an "
-                    "`abstract` or `base` class.",
+                    "deriving from final type `{0}`. Base type must be an "
+                    "`abstract` or `base` class",
```
Comment (zygoloid) [inline toolchain/check/handle_class.cpp]:
```suggestion
                    "deriving from final type `{0}`; base type must be an "
                    "`abstract` or `base` class",
```

File: toolchain/check/handle_function.cpp [hunk 6]
```diff
@@ -325,7 +325,7 @@ static auto BuildFunctionDecl(Context& context,
              context.GetBuiltinType(SemIR::BuiltinInstKind::IntType) &&
          return_type_id != context.GetTupleType({}))) {
       CARBON_DIAGNOSTIC(InvalidMainRunSignature, Error,
-                        "Invalid signature for `Main.Run` function. Expected "
+                        "invalid signature for `Main.Run` function. Expected "
                         "`fn ()` or `fn () -> i32`.");
```
Comment (zygoloid) [inline toolchain/check/handle_function.cpp]:
```suggestion
                        "invalid signature for `Main.Run` function; expected "
                        "`fn ()` or `fn () -> i32`");
```

File: toolchain/check/handle_literal.cpp [hunk 7]
```diff
@@ -126,7 +126,7 @@ static auto HandleIntOrUnsignedIntTypeLiteral(Context& context,
                                               IntId size_id) -> bool {
   if (!(context.ints().Get(size_id) & 3).isZero()) {
     CARBON_DIAGNOSTIC(IntWidthNotMultipleOf8, Error,
-                      "Bit width of integer type literal must be a multiple of "
+                      "bit width of integer type literal must be a multiple of "
                       "8. Use `Core.{0}({1})` instead.",
```
Comment (zygoloid) [inline toolchain/check/handle_literal.cpp]:
```suggestion
                      "8; use `Core.{0}({1})` instead",
```

File: toolchain/check/member_access.cpp [hunk 8]
```diff
@@ -528,8 +527,8 @@ auto PerformTupleAccess(Context& context, SemIR::LocId loc_id,
   auto tuple_type = context.types().TryGetAs<SemIR::TupleType>(tuple_type_id);
   if (!tuple_type) {
     CARBON_DIAGNOSTIC(TupleIndexOnANonTupleType, Error,
-                      "Type `{0}` does not support tuple indexing. Only "
-                      "tuples can be indexed that way.",
+                      "type `{0}` does not support tuple indexing. Only "
```
Comment (zygoloid) [inline toolchain/check/member_access.cpp]:
```suggestion
                      "type `{0}` does not support tuple indexing; only "
```

File: toolchain/parse/handle_import_and_package.cpp [hunk 9]
```diff
@@ -135,7 +135,7 @@ static auto VerifyInImports(Context& context, Lex::TokenIndex intro_token)
                         "`import` declarations must come after the `package` "
                         "declaration (if present) and before any other "
                         "entities in the file.");
```
Comment (zygoloid) [inline toolchain/parse/handle_import_and_package.cpp]:
```suggestion
                        "entities in the file");
```

File: toolchain/parse/handle_match.cpp [hunk 10]
```diff
@@ -12,7 +12,7 @@ namespace Carbon::Parse {
 // have a larger look at this code and how it produces parse errors. This may be
 // good to re-examine when someone is working on implementing match checking.
 static auto DiagnoseMatchParseTODO(Context& context) -> void {
-  CARBON_DIAGNOSTIC(MatchParseTODO, Error, "TODO: Improve match parsing.");
+  CARBON_DIAGNOSTIC(MatchParseTODO, Error, "tODO: Improve match parsing");
```
Comment (zygoloid) [inline toolchain/parse/handle_match.cpp]:
```suggestion
  CARBON_DIAGNOSTIC(MatchParseTODO, Error, "TODO: improve match parsing");
```

File: toolchain/parse/handle_match.cpp [hunk 11]
```diff
@@ -127,7 +127,7 @@ auto HandleMatchCaseLoopAfterDefault(Context& context) -> void {
 
   Lex::TokenKind kind = context.PositionKind();
   if (kind == Lex::TokenKind::Case or kind == Lex::TokenKind::Default) {
-    CARBON_DIAGNOSTIC(UnreachableMatchCase, Error, "Unreachable case; `{0}{1}",
+    CARBON_DIAGNOSTIC(UnreachableMatchCase, Error, "unreachable case; `{0}{1}",
                       Lex::TokenKind, std::string);
     context.emitter().Emit(*context.position(), UnreachableMatchCase, kind,
                            "` occurs after the `default`");
```
Comment (zygoloid) [inline toolchain/parse/handle_match.cpp]:
```suggestion
    CARBON_DIAGNOSTIC(UnreachableMatchCase, Error,
                      "unreachable case; `{0}` occurs after the `default`",
                      Lex::TokenKind);
    context.emitter().Emit(*context.position(), UnreachableMatchCase, kind);
```

**Review by zygoloid — APPROVED** [review]

**Comment by jonmeow** [comment]
For reference, handle_class.cpp is omitted here because of the separate #4327 (we were just doing related work at the same time)
